### ISSUE TITLE
`push qword` disassembled as `push dword`

### ISSUE THREAD

**Issue by nettrino** [Vector35/binaryninja-api#732]
In a statically-linked x86_64 binary,  binja recognizes `ff742458` as 
`push(zx.q([rsp + 0x58].d))` (linear disassembly `00400e29  ff742458  push    dword [rsp+0x58 {arg_8}]`)  whereas the appropriate instruction should be `push    qword ptr [rsp + 0x58]`.

Steps to reproduce:
Compile (using only the --static flag)
```
#include <stdio.h>

void foo(int x) {
    printf("Foo %d\n", x);
}

int main(int argc, char *argv[])
{
    if (argc > 2)
        foo(argc);
    else
        printf("boo\n");
    return 0;
}
```

Platform tested: Linux 4.4.0-62-generic #83-Ubuntu SMP x86_64 GNU/Linux
Binja version: 1.0.794-dev (build id a22d60e6)

**Comment by plafosse** [comment]
Thanks, this is a bug however it will be fixed by our upcoming switch to XED for doing x86 disassembly, so I am going to mark it as won't fix.  Keep track of #193 if you are interested.

**Comment by plafosse** [comment]
Fixed in 1.1.991
